java/src/com/ibm/streamsx/topology/TStream.java

```diff
@@ -89,6 +90,41 @@
      * @return Filtered stream
      */
     TStream<T> filter(Predicate<T> filter);
+    
+    /**
+     * Split a stream into {@code n} streams as specified by {@code splitter}.
+     * 
+     * <P>
+     * For each tuple on the stream, {@code splitter.applyAsInt(tuple)} is called.
+     * The return value {@code r} determines the destination stream:
+     * <pre>
+     * if r < 0 the tuple is discarded
+     * else it is sent to the stream at position (r % n) in the returned array.
+     * </pre>
+     * </P>
+     * 
+     * <P>
+     * Example of round robin splitting a stream of {@code String} into 5 channels.
```

Comment:
right, it makes that sort of thing possible.  I just noticed issue#114 which notes other inconveniences of the current scheme. We should probably warp over to that for further discussion on parallel().

Still interested in thoughts on whether split() should also offer a Keyable mode -- either a la issue#114 or implicit like today's parallel().  Maybe not.  Wonder if implements Keyable or issue#114 scheme really is a benefit over an explicit partitioning fn.  The splitter writer is also free to check (or know) if the type implements keyable and use that in their computation.  Mostly I was noting the dissimilarity in partitioning mechanisms between parallel() and split() and wondered if they should be more similar.